Add docker-compose setup for development

ChrisBr opened the pull request:
To make it easier to setup the whole development environment and avoid
to run a postgres / redis instance on your local development machine.

The docker-compose file will start three containers:
* postgres database (db)
* redis (cache)
* frontend (sidekiq, puma & webpacker)

We use the official postgres, redis and ruby docker images from DockerHub.

When starting the first time the containers, it is necessary to build the
frontend container with docker-compose build. After that, the containers
can get started with docker-compose up.

See updated README file for a more detailed explanation how to use
the docker setup.

Just as a suggestion as I prefer to run my development environments in isolation and keep my Laptop 'clean'. Feel free to close it if you don't like it.

It would be possible to automate the manual steps from README even further with a e.g. Rake task but this can be done in a separate PR.

Also I'm not sure about my changes to the ``Procfile`` if you use this in 'production' then these change are probably not possible.

lewispb commented:
Hey @ChrisBr thanks so much for the PR. I've setup Heroku review apps so we can check that the Procfile works. Could you please rebase master to test this works?

ChrisBr commented:
> I've setup Heroku review apps so we can check that the Procfile works. Could you please rebase master to test this works?

Rebased! However, I added now a Procfile.dev for development because starting the webpacker server in Heroku / production does not make sense (even if it would maybe work). I think this is cleaner now.

ChrisBr commented:
Dropped the environment for docker / development.

ChrisBr commented:
Thanks for merging :tada: :tada: :tada:

lewispb commented:
Sorry for the delay and thanks for contributing @ChrisBr !
